crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -427,6 +432,41 @@ impl World {
     }
 }
 
+/// Returns an [`Observer`] that watches for an input event, maps it to a new event type, and triggers
+/// that event.
+///
+/// # Examples
+///
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// # use bevy_ecs::observer::trigger_map;
+/// # let mut world = World::new();
+///
+/// #[derive(Event, Clone, Debug)]
+/// struct Foo(usize);
+///
+/// #[derive(Event, Debug)]
+/// struct Bar(bool);
+///
+/// #[derive(Event, Debug)]
+/// enum Combined {
+///     Foo(Foo),
+///     Bar(Bar),
+/// }
+///
+/// world.observe(trigger_map(Combined::Foo));
```

Comment:
IMO this should also trigger_map for `Combined::Bar` in this example: the point is to support either of these events.

Reply:
Yup good call